Classes/Utility/DeeplBackendUtility.php

```diff
@@ -270,7 +263,7 @@ public static function detectCurrentPage(ProcessingInstruction $processingInstru
                 (string)$processingInstruction->getProcessingTable(),
                 (int)$processingInstruction->getProcessingId()
             );
-            self::$currentPage = self::getPageRecord($pageId);
+            self::$currentPage = CurrentPage::fromArray(self::getPageRecord($pageId));
```

Comment:
Use a private method on the same class instead of having a public static factory method on the DTO object.